contracts/src/validity/OPSuccinctL2OutputOracle.sol

```diff
@@ -381,7 +392,7 @@ contract OPSuccinctL2OutputOracle is Initializable, ISemver {
     {
         // The proposer must be explicitly approved, or the zero address must be approved (permissionless proposing).
         require(
-            approvedProposers[msg.sender] || approvedProposers[address(0)],
```

Comment:
@ratankaliani I'm thinking of keeping `msg.sender` for `whenOptimistic`. What's your thoughts here? I'm really not sure.

Reply:
I'm fine to keep as is. You should add a note on `whenOptimistic` that it's not compatible with `OPSuccinctDisputeGame`.